### ISSUE TITLE
new version of PIL fails on JPEG, old version works

### ISSUE THREAD

**Issue by gabor** [python-pillow/Pillow#1328]
doing "PIL.Image.open(data)" on the attached JPEG works in old-version, fails in new-version. the commit that broke it is this:
https://github.com/python-pillow/Pillow/commit/a5683ab574ef6a356445b7a29c5dc2da2a1abc6f
(so, before this commit it works, with this commit it fails)

the image
![pil_bug](https://cloud.githubusercontent.com/assets/51989/8474800/1db438e0-20b3-11e5-8259-872cab411e64.jpg)

**Comment by hugovk** [comment]
That commit is from https://github.com/python-pillow/Pillow/pull/822, first released in Pillow 2.6.0.

**Comment by hugovk** [comment]
Can reproduce with Pillow 2.6.0, 2.8.0, 2.9.0 (Python 2.7, Windows 7). 
Pillow 2.5.0 works.
```python
from PIL import Image
im = Image.open("1328.jpg")
im.show()
```

**Comment by hugovk** [comment]
Uncommenting the traceback at line 2325 of Image.py gives this:

```
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\PIL\Image.py", line 2321, in open
    im = factory(fp, filename)
  File "C:\Python27\lib\site-packages\PIL\JpegImagePlugin.py", line 717, in jpeg_factory
    mpheader = im._getmp()
  File "C:\Python27\lib\site-packages\PIL\JpegImagePlugin.py", line 393, in _getmp
    return _getmp(self)
  File "C:\Python27\lib\site-packages\PIL\JpegImagePlugin.py", line 495, in _getmp
    raise SyntaxError("unsupported picture format in MPO")
SyntaxError: unsupported picture format in MPO
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\PIL\Image.py", line 2321, in open
    im = factory(fp, filename)
  File "C:\Python27\lib\site-packages\PIL\ImageFile.py", line 97, in __init__
    self._open()
  File "C:\Python27\lib\site-packages\PIL\ImImagePlugin.py", line 119, in _open
    raise SyntaxError("not an IM file")
SyntaxError: not an IM file
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\PIL\Image.py", line 2321, in open
    im = factory(fp, filename)
  File "C:\Python27\lib\site-packages\PIL\ImageFile.py", line 97, in __init__
    self._open()
  File "C:\Python27\lib\site-packages\PIL\ImtImagePlugin.py", line 44, in _open
    raise SyntaxError("not an IM file")
SyntaxError: not an IM file
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\PIL\Image.py", line 2321, in open
    im = factory(fp, filename)
  File "C:\Python27\lib\site-packages\PIL\ImageFile.py", line 97, in __init__
    self._open()
  File "C:\Python27\lib\site-packages\PIL\IptcImagePlugin.py", line 96, in _open
    tag, size = self.field()
  File "C:\Python27\lib\site-packages\PIL\IptcImagePlugin.py", line 76, in field
    raise SyntaxError("invalid IPTC/NAA file")
SyntaxError: invalid IPTC/NAA file
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\PIL\Image.py", line 2321, in open
    im = factory(fp, filename)
  File "C:\Python27\lib\site-packages\PIL\ImageFile.py", line 97, in __init__
    self._open()
  File "C:\Python27\lib\site-packages\PIL\MpegImagePlugin.py", line 71, in _open
    raise SyntaxError("not an MPEG file")
SyntaxError: not an MPEG file
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\PIL\Image.py", line 2321, in open
    im = factory(fp, filename)
  File "C:\Python27\lib\site-packages\PIL\ImageFile.py", line 97, in __init__
    self._open()
  File "C:\Python27\lib\site-packages\PIL\PcdImagePlugin.py", line 43, in _open
    raise SyntaxError("not a PCD file")
SyntaxError: not a PCD file
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\PIL\Image.py", line 2321, in open
    im = factory(fp, filename)
  File "C:\Python27\lib\site-packages\PIL\ImageFile.py", line 97, in __init__
    self._open()
  File "C:\Python27\lib\site-packages\PIL\PixarImagePlugin.py", line 46, in _open
    raise SyntaxError("not a PIXAR file")
SyntaxError: not a PIXAR file
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\PIL\Image.py", line 2321, in open
    im = factory(fp, filename)
  File "C:\Python27\lib\site-packages\PIL\ImageFile.py", line 97, in __init__
    self._open()
  File "C:\Python27\lib\site-packages\PIL\SpiderImagePlugin.py", line 114, in _open
    raise SyntaxError("not a valid Spider file")
SyntaxError: not a valid Spider file
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\PIL\Image.py", line 2321, in open
    im = factory(fp, filename)
  File "C:\Python27\lib\site-packages\PIL\ImageFile.py", line 97, in __init__
    self._open()
  File "C:\Python27\lib\site-packages\PIL\TgaImagePlugin.py", line 73, in _open
    raise SyntaxError("not a TGA file")
SyntaxError: not a TGA file
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\PIL\Image.py", line 2321, in open
    im = factory(fp, filename)
  File "C:\Python27\lib\site-packages\PIL\ImageFile.py", line 97, in __init__
    self._open()
  File "C:\Python27\lib\site-packages\PIL\XVThumbImagePlugin.py", line 47, in _open
    raise SyntaxError("not an XV thumbnail file")
SyntaxError: not an XV thumbnail file
Traceback (most recent call last):
  File "C:\stufftodelete\1328.py", line 2, in <module>
    im = Image.open("1328.jpg")
  File "C:\Python27\lib\site-packages\PIL\Image.py", line 2330, in open
    % (filename if filename else fp))
IOError: cannot identify image file '1328.jpg'
```

It's trying a bunch of image plugins, and none match. Focusing on the first:

```
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\PIL\Image.py", line 2321, in open
    im = factory(fp, filename)
  File "C:\Python27\lib\site-packages\PIL\JpegImagePlugin.py", line 717, in jpeg_factory
    mpheader = im._getmp()
  File "C:\Python27\lib\site-packages\PIL\JpegImagePlugin.py", line 393, in _getmp
    return _getmp(self)
  File "C:\Python27\lib\site-packages\PIL\JpegImagePlugin.py", line 495, in _getmp
    raise SyntaxError("unsupported picture format in MPO")
SyntaxError: unsupported picture format in MPO
```

Which leads to here:
https://github.com/python-pillow/Pillow/blob/master/PIL/JpegImagePlugin.py#L491-494

It raises because `mpentryattr['ImageDataFormat'] == 1`.

If the raise is commented, the image is shown.

Any suggestions on the best fix for this?

**Comment by aclark4life** [comment]
Ping @Feneric and/or @wiredfool

**Comment by wiredfool** [comment]
It looks like we completely barf on a bad MPO image, and it appears that there are many cameras out there that add the MPO headers to deal with things like thumbnails. 

I think we should catch a syntax error here: https://github.com/python-pillow/Pillow/blame/master/PIL/JpegImagePlugin.py#L722 separately, and issue some sort of warning that the image appears to be a malformed MPO file, and it will be interpreted as a base JPEG file. 

This would let any creative interpretations of the MPO spec pass as a base JPEG, and cover us in the cases where we don't actually cover all the possible values. (or, likely, both at the same time).

**Comment by Feneric** [comment]
Yeah, it looks like an MPO image that's bad in an unexpected way. I agree with @wiredfool that catching them at that point and treating them as JPEGs is a reasonable way to go.

**Comment by hugovk** [comment]
@gabor Would it be possible to include that image in the test suite, redistributable under a MIT-style licence? If not, do you have another problematic image that can be included?

**Comment by hugovk** [comment]
@wiredfool @Feneric Something like PR #1334?

**Comment by Feneric** [comment]
I haven't actually tried it, but yeah, that looks like a decent way to handle this type of image.

**Comment by wiredfool** [comment]
That's what I was thinking.

**Comment by gabor** [comment]
> @gabor Would it be possible to include that image in the test suite, redistributable under a MIT-style licence? If not, do you have another problematic image that can be included?

probably no :( i will have to check.. and regarding test-images, all of them are from a similar source, so the answer will be the same for all of them.

**Comment by hugovk** [comment]
Fixed via PR #1350 (which is the same as PR #1334 but with a redistributable image).
